File: Microsoft Office Developer Documentation Team/SharePoint-2010-Importing-Definitions-in-Visual-4046/[C#]-SharePoint 2010 Importing Site Definitions in Visual Studio 2010/C#/FilteredTaskSite/VisualWebPart1/VisualWebPart1UserControl.ascx.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using Microsoft.SharePoint;
using Microsoft.SharePoint.WebControls;
using Microsoft.SharePoint.Utilities;

namespace FilteredTaskSite.VisualWebPart1
{
    public partial class VisualWebPart1UserControl : UserControl
    {
       
        SPQuery query;
        SPWeb thisWeb;
        protected void Page_Load(object sender, EventArgs e)
        {
        thisWeb = SPContext.Current.Web;
        taskView.List = thisWeb.Lists["Project Tasks"];
        query = new SPQuery(taskView.List.DefaultView);
        query.ViewFields = "<FieldRef Name='Title' /><FieldRef Name='AssignedTo' /><FieldRef Name='DueDate' />";
        taskView.Query = query;
        }

        protected void filterDate_DateChanged(object sender, EventArgs e)
        {
            String camlQuery = "<Where><Leq><FieldRef Name='DueDate' />"
           + "<Value Type='DateTime'>"
           + SPUtility.CreateISO8601DateTimeFromSystemDateTime(DateTimeControl1.SelectedDate)
           + "</Value></Leq></Where>";
            query.Query = camlQuery;

        }
    }

}

File: Microsoft Office Developer Documentation Team/SharePoint-2010-Creating-Definitions-in-Visual-4023/[VB]-SharePoint 2010 Creating Site Definitions in Visual Studio 2010/VB/FilteredTaskSite/VisualWebPart1/VisualWebPart1UserControl.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports Microsoft.SharePoint
Imports Microsoft.SharePoint.WebControls
Imports Microsoft.SharePoint.Utilities

Partial Public Class VisualWebPart1UserControl
    Inherits UserControl
    Dim query As SPQuery
    Dim thisWeb As SPWeb


    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
        thisWeb = SPContext.Current.Web
        taskView.List = thisWeb.Lists("Project Tasks")
        query = New SPQuery(taskView.List.DefaultView)
        query.ViewFields = "<FieldRef Name='Title' /><FieldRef Name='AssignedTo' /><FieldRef Name='DueDate' />"
        taskView.Query = query
    End Sub
    Sub filterDate_DateChanged(ByVal sender As Object, ByVal e As EventArgs)

        Dim camlQuery As String
        camlQuery = "<Where><Leq><FieldRef Name='DueDate' />" _
               + "<Value Type='DateTime'>" _
               + SPUtility.CreateISO8601DateTimeFromSystemDateTime(DateTimeControl1.SelectedDate) _
               + "</Value></Leq></Where>"
        query.Query = camlQuery

    End Sub
End Class
